libqtile/widget/widgetbox.py

```diff
@@ -0,0 +1,169 @@
+    work.
+
+    Widgets in the box alsos remain accessible via command interfaces.
+
+    Widgets can only be added to the box via the configuration file. The widget
+    is conigured by adding widgets to the "widgets" parameter as follows:
+
+        widget.WidgetBox(widgets=[
+            widget.TextBox(text="This widget is in the box"),
+            widget.Memory(),
+            ]),
+    """
+    orientations = base.ORIENTATION_HORIZONTAL
+    defaults = [
+        (
+            "font",
+            "sans",
+            "Text font"
+        ),
+        (
+            "fontsize",
+            None,
+            "Font pixel size. Calculated if None."
+        ),
+        (
+            "fontshadow",
+            None,
+            "font shadow color, default is None(no shadow)"
+        ),
+        (
+            "foreground",
+            "#ffffff",
+            "Foreground colour."
+        ),
+        (
+            "close_button_location",
+            "left",
+            "Location of close button when box open ('left' or 'right')"
+        ),
+        (
+            "text_closed",
+            "[<]",
+            "Text when box is closed"
+        ),
+        (
+            "text_open",
+            "[>]",
+            "Text when box is open"
+        ),
+    ]
+
+    def __init__(self, widgets=list(), **config):
+        base._Widget.__init__(self, bar.CALCULATED, **config)
+        self.add_defaults(WidgetBox.defaults)
+        self.box_is_open = False
+        self.widgets = widgets
+        self.add_callbacks({"Button1": self.toggle_box})
+
+        if self.close_button_location not in ["left", "right"]:
+            val = self.close_button_location
+            msg = "Invalid value for 'close_button_location': {}".format(val)
+            logger.warning(msg)
+            self.close_button_location = "left"
+
+    def _configure(self, qtile, bar):
+        base._Widget._configure(self, qtile, bar)
+        self.set_box_label()
+
+        for idx, i in enumerate(self.widgets):
+            if i.configured:
+                i = i.create_mirror()
+                self.widgets[idx] = i
+            qtile.register_widget(i)
+            i._configure(qtile, bar)
+
+    def calculate_length(self):
+        return self.layout.width
+
+    def set_box_label(self):
+        self.layout = self.drawer.textlayout(
+            self.text_open if self.box_is_open else self.text_closed,
+            self.foreground,
+            self.font,
+            self.fontsize,
+            self.fontshadow,
+            markup=False,
+        )
+
+    def toggle_widgets(self):
+        for widget in self.widgets:
+            try:
+                self.bar.widgets.remove(widget)
+            except ValueError:
+                continue
+        try:
+            index = self.bar.widgets.index(self)
+        except ValueError:
+            logger.warning("Can't locate item in bar")
```

Comment:
I think this try block can be removed, just leaving the `index =` line. I don't think this error can get raised (under normal usage), and if it does, then we want whatever is trying to toggle the widgets to stop trying, rather than to simply log and carry on. Indeed, as is, line 136 will error because `index` doesn't exist if this error is raised.